src/main/java/io/github/syst3ms/skriptparser/expressions/CondExprEmpty.java

```diff
@@ -0,0 +1,64 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import java.util.Arrays;
+
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.lang.base.ConditionalExpression;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+
+/**
+ * Check if a given string or list is empty.
+ *
+ * @name Empty
+ * @pattern %strings% (is|are)[1:( not|n't)] empty
+ * @pattern %objects% is[1:( not|n't)] [a[n]] empty list
+ * @since ALPHA
+ * @author Olyno
+ */
+public class CondExprEmpty extends ConditionalExpression {
+
+    static {
+        Parser.getMainRegistration().addExpression(CondExprEmpty.class,
+            Boolean.class,
+            true,
+            2,
+            "%strings% (is|are)[1:( not|n't)] empty",
+            "%objects% is[1:( not|n't)] [a[n]] empty list"
+        );
+    }
+
+    private Expression<?> expr;
+    private boolean comparingList;
+
+    @Override
+    public boolean init(Expression<?>[] expressions, int matchedPattern, ParseContext parseContext) {
+        expr = (Expression<?>) expressions[0];
```

Comment:
This is a redundant cast.

Reply:
How could I improve that?

Reply:
Just remove the cast. expression[0] is by definition already an instance of Expression<?>.